components/gantry/collectors_test.go

```diff
@@ -0,0 +1,103 @@
+package gantry
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	clk "github.com/benbjohnson/clock"
+	"github.com/edaniels/golog"
+	pb "go.viam.com/api/component/gantry/v1"
+	"go.viam.com/test"
+	"go.viam.com/utils/protoutils"
+
+	"go.viam.com/rdk/data"
+	"go.viam.com/rdk/resource"
+	tu "go.viam.com/rdk/testutils"
+)
+
+type collectorFunc func(resource interface{}, params data.CollectorParams) (data.Collector, error)
+
+const componentName = "gantry"
+
+var floatList = []float64{1.0, 2.0, 3.0}
+
+func TestGantryCollectors(t *testing.T) {
+	tests := []struct {
+		name      string
+		params    data.CollectorParams
+		collector collectorFunc
+		expected  map[string]any
+	}{
+		{
+			name: "Length collector should write a lengths response",
+			params: data.CollectorParams{
+				ComponentName: componentName,
+				Interval:      time.Second,
+				Logger:        golog.NewTestLogger(t),
+			},
+			collector: newLengthsCollector,
+			expected: toProtoMap(pb.GetLengthsResponse{
+				LengthsMm: scaleMetersToMm(floatList),
+			}),
+		},
+		{
+			name: "End position collector should write a list of positions",
+			params: data.CollectorParams{
+				ComponentName: componentName,
+				Interval:      time.Second,
+				Logger:        golog.NewTestLogger(t),
+			},
+			collector: newPositionCollector,
+			expected: toProtoMap(pb.GetPositionResponse{
+				PositionsMm: scaleMetersToMm(floatList),
+			}),
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			mockClock := clk.NewMock()
+			buf := tu.MockBuffer{}
+			tc.params.Clock = mockClock
+			tc.params.Target = &buf
+
+			gantry := newGantry(componentName)
+			col, err := tc.collector(gantry, tc.params)
+			test.That(t, err, test.ShouldBeNil)
+
+			defer col.Close()
+			col.Collect()
+			mockClock.Add(1 * time.Second)
```

Comment:
Can you make time.Second a const or var named `captureInterval` or `interval` or something? It's referenced a few places that need to match

Reply:
Likewise for other tests with this pattern